ui/src/main/scripts/plugins/e2e/ts/e2ePlugin.ts

```diff
@@ -26,5 +26,44 @@ module E2E {
     // Doesn't link an empty block
   }]);
 
+  _module.filter('hkDuration', function (limitToFilter, numberFilter) {
+    let milliRange =  1000;
+    let secondRange = 1000 * milliRange;
+    let minuteRange =   60 * secondRange;
+    let hourRange =     60 * minuteRange;
+    let dayRange =      24 * hourRange;
+    let weekRange =      7 * dayRange;
+    let yearRange =    365 * dayRange;
+
+    return function (input, hideSecondary) {
+      if (!isNaN(input)) {
+        if (input > yearRange) {
+          let weeks = input / weekRange;
+          return Math.floor(weeks / 52) + 'y ' + (hideSecondary ? '' : numberFilter(Math.floor(weeks % 52), 0) + 'w');
```

Comment:
Yes, that was the precision we agreed on at https://issues.jboss.org/browse/HWKAPM-602